File: Two_Pointers/011.Container-With-Most-Water/011.Container-With-Most-Water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int max = 0;
        
        while (left < right) {
            max = Math.max(max, (right - left) * Math.min(height[left], height[right]));
            if (height[left] <= height[right]) {
                left += 1;
            } else {
                right -= 1;
            }
        }
        return max;
    }
}

File: Two_Pointers/011.Container-With-Most-Water/011.Container-With-Most-Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0;
        right = len(height) - 1;
        
        result = 0;
        while left < right:
            currentHeight = min(height[left], height[right]);
            currentWidth = right - left;
            result = max(result, currentHeight * currentWidth);
            if height[left] < height[right]:
                left += 1;
            else:
                right -= 1;
        return result;
